PAINTROID-242 added Paintroids dependencies to Catroid

ldirry opened the pull request:
Children dependencies don't get included, when the parent is included as .aar dependecy.  I also added the exifinterface dependecy from Paintroid because this could also be a potential ClassNotFound bug.
related solution (https://stackoverflow.com/questions/56989569/how-to-fix-java-lang-noclassdeffounderror-in-android-aar-project)

https://jira.catrob.at/browse/PAINTROID-242
https://jira.catrob.at/browse/CATROID-1039

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

wslany reviewed (changes requested):
Thanks! However, it seems that our exif related test now fails consistently, even though it normally does not. Could you check whether there's some issue due to now conflicting libraries or something along these lines? See, e.g., the stacktrace from the jenkins job of the present PR here https://jenkins.catrob.at/job/Catroid/job/PR-4130/2/testReport/org.catrobat.catroid.test.utiltests/RemoveExifDataTest/All___1___Instrumented_Unit_Tests___testRemoveExifData/
```
junit.framework.AssertionFailedError
at junit.framework.Assert.fail(Assert.java:48)
at junit.framework.Assert.assertTrue(Assert.java:20)
at junit.framework.Assert.assertTrue(Assert.java:27)
at org.catrobat.catroid.test.utiltests.RemoveExifDataTest.testRemoveExifData(RemoveExifDataTest.java:75)
at java.lang.reflect.Method.invoke(Native Method)
at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at androidx.test.ext.junit.runners.AndroidJUnit4.run(AndroidJUnit4.java:104)
at org.catrobat.catroid.runner.AndroidPackageRunner.runChild(AndroidPackageRunner.java:101)
at org.catrobat.catroid.runner.AndroidPackageRunner.runChild(AndroidPackageRunner.java:48)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at org.junit.runners.Suite.runChild(Suite.java:128)
at org.junit.runners.Suite.runChild(Suite.java:27)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at org.junit.runners.Suite.runChild(Suite.java:128)
at org.junit.runners.Suite.runChild(Suite.java:27)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
at androidx.test.internal.runner.TestExecutor.execute(TestExecutor.java:56)
at androidx.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:388)
at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:2145)
```

Compare to the normally green runs for the develop branch https://jenkins.catrob.at/job/Catroid/job/develop/test_results_analyzer/
![image](https://user-images.githubusercontent.com/1747888/115333032-408d4380-a199-11eb-8236-f3f13f84ed84.png)

ldirry commented:
Ok thats interesting, the exifinterface library is already included via another dependency (couldn't figure out which dependency). And there seemed to be some version conflicts, when I added the the newest version (1.3.2) to the build.gradle. 

In the older version the `setAttribute()` seems to allow empty strings for some tags/attributes (however, not in 1.3.2 anymore). So I removed the 1.3.2 dependency from the build.gradle again and added a testcase where the ExifInterface gets called from pocket paint. Rewrote it in kotlin.

dandrawes reviewed (approved):
Well done!

wslany commented on catroid/src/androidTest/java/org/catrobat/catroid/test/utiltests/ExifDataTest.kt:
```diff
@@ -0,0 +1,89 @@
+    @Test
+    fun testRemoveExifData() {
+        var exif = ExifInterface(cacheFile.absolutePath)
+        Assert.assertFalse(exif.getAttribute(ExifInterface.TAG_ARTIST)!!.isEmpty())
+        Assert.assertFalse(exif.getAttribute(ExifInterface.TAG_DATETIME)!!.isEmpty())
```
Very good, the !! is used correctly here.

wslany commented:
Thanks!
